system/jlib/jsocket.cpp

```diff
@@ -6794,3 +6805,130 @@ int wait_write_multiple(UnsignedArray &socks,       //IN   sockets to be checked
 {
     return wait_multiple(false, socks, timeoutMS, readySocks);
 }
+
+
+class CWhiteListHandler : public CSimpleInterfaceOf<IWhiteListHandler>
+{
+    struct PairHasher
+    {
+        template <class T1, class T2>
+        std::size_t operator () (std::pair<T1, T2> const &pair) const
+        {
+            std::size_t h1 = std::hash<T1>()(pair.first);
+            std::size_t h2 = std::hash<T2>()(pair.second);
+            return h1 ^ h2;
+        }
+    };
+
+    using WhiteListHT = std::unordered_set<std::pair<std::string, unsigned __int64>, PairHasher>;
+    WhiteListPopulateFunction populateFunc;
```

Comment:
The use of callbacks seems unnecessarily complex

Reply:
Really all of this should be in the environment.* (as comment says somewhere), but I can't make libenvironment.so (that's true before this change too),
but that causes a circular dependency..